src/core/menuinst.cpp

```diff
@@ -0,0 +1,455 @@
+#include <string>
+#include <iostream>
+#include <codecvt>
+#include <regex>
+#include <reproc++/run.hpp>
+
+#include "mamba/core/context.hpp"
+#include "mamba/core/transaction_context.hpp"
+#include "mamba/core/shell_init.hpp"
+#include "mamba/core/environment.hpp"
+#include "mamba/core/virtual_packages.hpp"
+#include "mamba/core/mamba_fs.hpp"
+#include "mamba/core/util.hpp"
+
+#ifdef _WIN32
+#include <windows.h>
+#include <commctrl.h>
+#include <imagehlp.h>
+#include <objbase.h>
+#include <objidl.h>
+#include <shlobj.h>
+#endif
+
+namespace mamba
+{
+    namespace detail
+    {
+        std::string get_formatted_env_name(const fs::path& target_prefix)
+        {
+            std::string name = env_name(target_prefix);
+            if (name.find_first_of("\\/") != std::string::npos)
+            {
+                return "";
+            }
+            return name;
+        }
+    }
+
+#ifdef _WIN32
+    namespace win
+    {
+        /*
+         * Create a shortcut on a Windows Machine (using COM)
+         * Place the shortcut in `programs` directory to make it appear in the
+         * startmenu.
+         *
+         * Args:
+         *   path: Path to the executable
+         *   description: string that is displayed
+         *   filename: Link destination filename
+         *   arguments: args to the executable (optional)
+         *   work_dir: workdir for the executable
+         *   icon_path: path to an .ico file
+         *   icon_index: index for icon
+         */
+        void create_shortcut(const fs::path& path,
+                             const std::string& description,
+                             const fs::path& filename,
+                             const std::string& arguments,
+                             const fs::path& work_dir,
+                             const fs::path& icon_path,
+                             int icon_index)
+        {
+            IShellLink* pShellLink = nullptr;
+            IPersistFile* pPersistFile = nullptr;
+
+            HRESULT hres;
+            LOG_WARNING << "Creating shortcut with "
```

Comment:
why a warning and not info log level ?
a warning should be reserved to something that went wrong but not preventing to continue the operation (using a fallback for example) or to something that represent a potential risk of failure (using experimental mode, etc.)

Reply:
just for debug